gpflow/base.py

```diff
@@ -225,9 +232,7 @@ def _run_op(a, *args, **kwargs):
 tf.register_tensor_conversion_function(Parameter, lambda x, *args, **kwds: x.read_value())
 
 
-def _verified_value(value: VariableData, dtype: Optional[DType] = None) -> np.ndarray:
-    if isinstance(value, tf.Variable):
-        return value
+def _cast_to_dtype(value: VariableData, dtype: Optional[DType] = None) -> tf.Tensor:
```

Comment:
```suggestion
def _cast_to_default_dtype(value: VariableData, dtype: Optional[DType] = None) -> tf.Tensor:
```

Reply:
It's not the default_dtype if the `dtype` argument is actually passed in. We could add a docstring, if you want, though it's an internal (module-private) function and very short so I thought that'd be overkill.